genie-agent/src/main/java/com/netflix/genie/agent/execution/services/ArchivalService.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.execution.services;
+
+import com.netflix.genie.agent.execution.exceptions.ArchivalException;
+
+import java.io.File;
+import java.net.URI;
+
+/**
+ * Service to archive a local file to a target location URI.
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+public interface ArchivalService {
+
+    /**
+     * Archive file to a target location.
+     *
+     * @param file              file/dir to archive
+     * @param targetLocationURI target location uri for the file
+     * @throws ArchivalException if archival fails
+     */
+    void archive(final File file, final URI targetLocationURI) throws ArchivalException;
```

Comment:
As I mentioned in @mprimi review for #784 I think we should be targeting using `java.nio.*` classes going forward instead of the older `java.io.*` classes where possible.